### PR TITLE
Add Nemotron HF Support

### PR DESCRIPTION
**Pull request by suiyoubi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/nemotron/modeling_nemotron.py [hunk 1]
```diff
@@ -0,0 +1,1560 @@
...
+class NemotronLayerNorm(nn.Module):
+    def __init__(self, hidden_size, normalization='layernorm1p', eps=1e-6):
+        super().__init__()
+        self.normalization = normalization
+        if normalization == 'layernorm1p':
+            self.ln = LayerNorm1P(hidden_size, bias=True, eps=eps)
+        elif normalization == 'layernorm':
+            self.ln = nn.LayerNorm(hidden_size, bias=True, eps=eps)
+        else:
+            raise ValueError(f'Unsupported layernorm type: {normalization}')
```
Comment (ArthurZucker) [inline src/transformers/models/nemotron/modeling_nemotron.py]: This if else is the codepath I am talking about! If all model use LayerNorm1P, use layernorm1p only. 
TLDR we'd rather have a mapping with layer norm name - layer norm class. But if all models use the same we remove the one that does not have pretrained checkpoints associated to it!
